instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/jmx/BeanFinder.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.runtimemetrics.jmx;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.management.MBeanServer;
+import javax.management.MBeanServerFactory;
+import javax.management.ObjectName;
+
+/**
+ * A class responsible for finding MBeans that match metric definitions specified by a set of
+ * MetricDefs.
+ */
+class BeanFinder {
+
+  private final MetricRegistrar registrar;
+  private MetricConfiguration conf;
+  private final ScheduledExecutorService exec = Executors.newSingleThreadScheduledExecutor();
+  private final long exportInterval;
+  private final long maxDelay;
+  private long delay = 1000; // number of milliseconds until first attempt to discover MBeans
+
+  BeanFinder(MetricRegistrar registrar, ConfigProperties configProperties) {
+    this.registrar = registrar;
+    this.exportInterval = configProperties.getLong("otel.metric.export.interval", 60000);
+    this.maxDelay = Math.max(60000, exportInterval);
+  }
+
+  void discoverBeans(MetricConfiguration conf) {
+    this.conf = conf;
+
+    exec.schedule(
```

Comment:
Can we just call `scheduleAtFixedRate` instead of repeatedly scheduling a single task?

Reply:
We know that in practice most of the MBeans will be created relatively quickly when the application or framework starts up, and most likely reach a steady state soon afterwards, so we want to run the discovery with decreasing frequency.

Reply:
Ah I missed that the delay is backing off with each run.